src/C++/SSLSocketInitiator.cpp

```diff
@@ -433,6 +522,33 @@ void SSLSocketInitiator::onTimeout( SocketConnector& )
     i->second->onTimeout();
 }
 
+void SSLSocketInitiator::disconnectPendingSSLHandshakesThatTakeTooLong(time_t now) {
```

Comment:
This method will go through all connections in the pendingSSLHandshake state, and if the handshake stated over 10 seconds ago, it will remove the socket from monitor, close the socket, and mark the session as disconnected, so that the engine will try to reconnect.